### PR TITLE
Deprecate unused initializers

### LINKED ISSUE #4093: Deprecate unused initializers
Ember Data has 4 initializers that it supports for legacy reasons but otherwise do nothing. https://github.com/emberjs/data/blob/master/tests/ember-data-initializers.js#L18-L40

@rwjblue [suggested](https://github.com/emberjs/data/pull/4088#issuecomment-172663110) that it may be possible to deprecate them by "hook[ing] into the DAG map that is used for initializers and see if anyone depends on it"

### PR DESCRIPTION
**Pull request by bmac**
Closes #4093 

- [ ] Deprecation RFC https://github.com/emberjs/rfcs/pull/181

### CODE REVIEW

**Comment by stefanpenner** [comment]
👍

File: app/initializers/data-adapter.js [hunk 1]
```diff
@@ -10,5 +10,5 @@ import Ember from 'ember';
 export default {
   name: 'data-adapter',
   before: 'store',
-  initialize: Ember.K
```
Comment (locks) [inline app/initializers/data-adapter.js]: Rebase needed?

**Comment by bmac** [comment]
rebased @locks

File: addon/-private/instance-initializers/initialize-store-service.js [hunk 2]
```diff
@@ -1,12 +1,53 @@
...
   // Eagerly generate the store so defaultStore is populated.
   container.lookup('service:store');
+
+  let initializers = application.application.constructor.initializers;
+  deprecateOldEmberDataInitializers(initializers)
+}
+
+
+
+let deprecatedInitializerNames = ['data-adapter', 'injectStore', 'transforms', 'store'];
```
Comment (bmac) [inline addon/-private/instance-initializers/initialize-store-service.js]: `store` should not be included in this list.

Comment (bmac) [inline addon/-private/instance-initializers/initialize-store-service.js]: Nevermind. This code is correct. I got `store` confused with the `ember-data` initializer.

**Comment by bmac** [comment]
I was given feedback that this should go through an RFC so there is an official document on why this has been deprecated.

https://github.com/emberjs/rfcs/pull/181

**Pull request merged**